blockchain/chain/src/store/chain_store.rs

```diff
@@ -492,30 +493,24 @@ where
 
         // Walks over tipset and historical data, sending all blocks visited into the car writer.
         Self::walk_snapshot(tipset, recent_roots, skip_old_msgs, |cid| {
-            let block = self.blockstore().get(&cid)?.ok_or_else(|| {
-                if skip_old_msgs {
-                    anyhow::anyhow!("Cid {cid} not found in blockstore")
-                } else {
-                    anyhow::anyhow!("Cid {cid} not found in blockstore. \
-                                    Exporting a full snapshot is only possible when the node is initialised with a full one. \
-                                    Consider exporting a lightweight snapshot, e.g. skip the old messages.")
-                }
-            })?;
-
-            // XXX: * If cb can return a generic type, deserializing would remove need to clone.
-            // Ignore error intentionally, if receiver dropped, error will be handled below
-            let async_handle = tokio::runtime::Handle::current();
-            let block_clone = block.clone();
             let tx_clone = tx.clone();
-            tokio::task::block_in_place(move || {
-                async_handle.block_on(async {
-                    tx_clone.send_async((cid, block_clone)).await.expect("failed sending block for export");
-                });
-            });
-
-            Ok(block)
-        })
-        .await?;
+            async move {
+                let block = self.blockstore().get(&cid)?.ok_or_else(|| {
+                    if skip_old_msgs {
+                        Error::Other("Cid {cid} not found in blockstore".to_string())
+                    } else {
```

Comment:
I think we should get rid of `skip_old_msgs`. Lotus is also moving away from "fat" snapshots ("fat" snapshots were used by block-explorers in the past but they can download the message receipts in other ways).

Not sure if removing the flag is appropriate for this PR or if it should be in a separate PR.

Reply:
I see, can remove it but removing `skip_old_msgs` would touch other modules and probably make this PR big. I can do that in a follow up PR.